e2e/ctl_v3_lease_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); // you may not use this file except in compliance with the License.  // You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestCtlV3Lease(t *testing.T)          { testCtl(t, leaseTest) }
+func TestCtlV3LeaseNoTLS(t *testing.T)     { testCtl(t, leaseTest, withCfg(configNoTLS)) }
+func TestCtlV3LeaseClientTLS(t *testing.T) { testCtl(t, leaseTest, withCfg(configClientTLS)) }
+func TestCtlV3LeasePeerTLS(t *testing.T)   { testCtl(t, leaseTest, withCfg(configPeerTLS)) }
+func TestCtlV3LeaseTimeout(t *testing.T)   { testCtl(t, leaseTest, withDialTimeout(0)) }
+
+func leaseTest(cx ctlCtx) {
+	defer close(cx.errc)
+
+	tests := []struct {
+		ttl    int
+		put    kv
+		revoke bool
+		gets   []kv
+	}{
+		{ // live lease
+			10,
+			kv{"key1", "val"},
+			false,
+			[]kv{{"key1", "val"}},
+		},
+		{ // revoked
+			10,
+			kv{"key2", "val"},
+			true,
+			nil,
+		},
+	}
+
+	for i, tt := range tests {
+		leaseID, err := ctlV3LeaseCreate(cx, tt.ttl)
+		if err != nil {
+			if cx.dialTimeout > 0 && !isGRPCTimedout(err) {
+				cx.t.Fatalf("leaseTest #%d: ctlV3LeaseCreate error (%v)", i, err)
+			}
+		}
+		if err := ctlV3Put(cx, tt.put.key, tt.put.val, leaseID); err != nil {
+			if cx.dialTimeout > 0 && !isGRPCTimedout(err) {
+				cx.t.Fatalf("leaseTest #%d: ctlV3Put error (%v)", i, err)
+			}
+		}
+		if tt.revoke {
+			if err := ctlV3LeaseRevoke(cx, leaseID); err != nil {
+				if cx.dialTimeout > 0 && !isGRPCTimedout(err) {
+					cx.t.Fatalf("leaseTest #%d: ctlV3LeaseRevoke error (%v)", i, err)
+				}
+			}
+			if err := ctlV3Get(cx, []string{tt.put.key}); err != nil {
```

Comment:
it looks like the Get below this one will do the same check again-- is this Get redundant?